web/src/main/java/org/cbioportal/web/StudyController.java

```diff
@@ -98,9 +99,9 @@ public ResponseEntity<List<CancerStudy>> getAllStudies(
         @ApiParam("Name of the property that the result list is sorted by")
         @RequestParam(required = false) StudySortBy sortBy,
         @ApiParam("Direction of the sort")
-        @RequestParam(defaultValue = "ASC") Direction direction,
-        @ApiParam(hidden = true, required = false)
-        @RequestParam(required = false) Authentication authentication) {
+        @RequestParam(defaultValue = "ASC") Direction direction
+    ) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
Problem:
- The `authentication` param _without_ annotations generates faulty/messy Swagger documentation
- The `authentication` param _with_  annotations results in all studies being marked with  `readPermission=true`
Maybe better to move the authentication parameter and instead fetch it from the security context?
@dippindots @JREastonMarks

Reply:
This looks ok to me, @jagnathan is there any potential issue if we merge this pr as is?

Reply:
@dippindots in the file org/mskcc/cbio/portal/util/internal/AccessControlImpl.java, there is some more complex logic for getting the authentication object. Do we need to follow that logic?